### PR TITLE
Enable local file sync on android

### PR DESCRIPTION
**Pull request by zoli**
# Description

Using the interface added on https://github.com/johannesjo/super-productivity-android/pull/25 now we can have local file sync on android.

## Issues Resolved

https://github.com/johannesjo/super-productivity-android/issues/13 and https://github.com/johannesjo/super-productivity/issues/2001

### CODE REVIEW

**Comment by zoli** [comment]
This should get merged after https://github.com/johannesjo/super-productivity-android/pull/25

File: src/app/features/config/form-cfgs/sync-form.const.ts [hunk 1]
```diff
@@ -47,11 +47,32 @@ export const SYNC_FORM: ConfigFormSection<SyncConfig> = {
             ? []
             : [{ label: SyncProvider.GoogleDrive, value: SyncProvider.GoogleDrive }]),
           { label: SyncProvider.WebDAV, value: SyncProvider.WebDAV },
-
-          ...(IS_ELECTRON
-            ? [{ label: SyncProvider.LocalFile, value: SyncProvider.LocalFile }]
-            : []),
+          { label: SyncProvider.LocalFile, value: SyncProvider.LocalFile },
```
Comment (johannesjo) [inline src/app/features/config/form-cfgs/sync-form.const.ts:50]: We still need to exclude the web platform, so we might do it like this:

```suggestion
      ...((IS_ELECTRON || IS_ANDROID_WEB_VIEW)
            ? [{ label: SyncProvider.LocalFile, value: SyncProvider.LocalFile }]
            : []),
```

**Comment by johannesjo** [comment]
Great job once more! Thanks a lot. I just have the one remark.

File: src/app/features/config/form-cfgs/sync-form.const.ts (hunk 1 above)
Comment (johannesjo) [reply src/app/features/config/form-cfgs/sync-form.const.ts:50]: Thinking about it it, I wondered if we could also check here if the android version has the required interfaces available and only show the option then?

Comment (zoli) [reply src/app/features/config/form-cfgs/sync-form.const.ts:50]: Yes you are right about both points.

Comment (zoli) [reply src/app/features/config/form-cfgs/sync-form.const.ts]: Resolved with https://github.com/johannesjo/super-productivity/pull/2195/commits/03eab3c06f3aa3883d17ce04fcdbf24219d9e18d I guess.

File: src/app/features/config/form-cfgs/sync-form.const.ts [hunk 2]
```diff
@@ -47,11 +48,36 @@ export const SYNC_FORM: ConfigFormSection<SyncConfig> = {
...
           { label: SyncProvider.WebDAV, value: SyncProvider.WebDAV },
-
-          ...(IS_ELECTRON
-            ? [{ label: SyncProvider.LocalFile, value: SyncProvider.LocalFile }]
-            : []),
+          ...((IS_ELECTRON || IS_ANDROID_WEB_VIEW) &&
+          androidInterface.grantFilePermissionWrapped &&
+          androidInterface.isGrantedFilePermission
+            ? []
+            : [{ label: SyncProvider.LocalFile, value: SyncProvider.LocalFile }]),
```
Comment (johannesjo) [inline src/app/features/config/form-cfgs/sync-form.const.ts:55]: If I am not mistaken this should be:
```suggestion
          ...((IS_ELECTRON || 
          (IS_ANDROID_WEB_VIEW &&
          androidInterface.grantFilePermissionWrapped &&
          androidInterface.isGrantedFilePermission))
            ? [{ label: SyncProvider.LocalFile, value: SyncProvider.LocalFile }]
            : []),
```

Comment (zoli) [reply src/app/features/config/form-cfgs/sync-form.const.ts]: Yes, I got a bit hasty. Fixed it.

**Comment by johannesjo** [comment]
Awesome! Thank you!

**Pull request merged**